Wikipedia/Code/WKWebViewWithSettableInputViews.swift

```diff
@@ -89,24 +89,53 @@ class WKWebViewWithSettableInputViews: WKWebView {
     }
 }
 
-private typealias ClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void
-private typealias BlockType =  @convention(block) (Any, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void
-private extension WKWebViewWithSettableInputViews {
-    private func setKeyboardRequiresUserInteraction(_ value: Bool) {
+//https://stackoverflow.com/questions/32449870/programmatically-focus-on-a-form-in-a-webview-wkwebview
+
+typealias OldClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Any?) -> Void
+typealias NewClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Bool, Any?) -> Void
+
+extension WKWebViewWithSettableInputViews {
+
+    func setKeyboardRequiresUserInteraction( _ value: Bool) {
         guard let WKContentView: AnyClass = NSClassFromString("WKContentView") else {
-            DDLogError("Could not get class")
+            print("keyboardDisplayRequiresUserAction extension: Cannot find the WKContentView class")
             return
         }
-        let sel = sel_getUid("_startAssistingNode:userIsInteracting:blurPreviousNode:changingActivityState:userObject:")
-        guard let method = class_getInstanceMethod(WKContentView, sel) else {
-            DDLogError("Could not get method")
-            return
+        // For iOS 11.3, *
```

Comment:
good catch on the methods signatures!

Reply:
I can't take credit, it's a stack overflow answer linked above

Reply:
i did notice this https://stackoverflow.com/a/55344531 example DRY's the swift up a bit - might be nice

Reply:
but i don't feel strongly about it :)